common/lib/xmodule/xmodule/tests/test_library_content.py

```diff
@@ -348,6 +353,62 @@ def _change_count_and_refresh_children(self, count):
         assert len(selected) == count
         return selected
 
+    @ddt.data(
+        # User resets selected children with reset button on capa blocks
+        (True, 1),
+        # User resets selected children without reset button on capa blocks
+        (False, 1),
+    )
+    @ddt.unpack
+    def test_reset_selected_children_capa_blocks(self, show_reset_button, max_count):
+        """
+        Tests that a content block is reset when
+        show_reset_button is True
+        """
+        self.lc_block.show_reset_button = show_reset_button
+        self.lc_block.max_count = max_count
+        # Add some capa blocks
+        self._create_capa_problems()
+        # Filter out children without capa_type of optionresponse
+        # at the end of the day we are left with only capa blocks
+        self.lc_block.capa_type = "optionresponse"
```

Comment:
It would be enough to have a single test here - this test block has 4 HTML XBlocks and 4 Problem XBlocks. Therefore, if we ensure that the `reset_problem` has been called `len(self.problem_types` times, then it means that this is working correctly. Let's describe this in the test's description after implementing this check.